networkx/algorithms/tests/test_simple_paths.py

```diff
@@ -259,3 +259,151 @@ def test_bidirectional_dijkstra_no_path():
+    paths = nx.bidirectional.all_simple_paths(G,0,3)
+    assert_equal(list(list(p) for p in paths),[[0,1,2,3]])
+
+def test_bidirectional_all_simple_paths_cutoff():
+    G = nx.complete_graph(4)
+    paths = nx.bidirectional.all_simple_paths(G,0,1,cutoff=1)
+    assert_equal(list(list(p) for p in paths),[[0,1]])
+    paths = nx.bidirectional.all_simple_paths(G,0,1,cutoff=2)
+    assert_equal(list(list(p) for p in paths),[[0,1],[0,2,1],[0,3,1]])
+
+def test_bidirectional_all_simple_paths_multigraph():
+    G = nx.MultiGraph([(1,2),(1,2)])
+    paths = nx.bidirectional.all_simple_paths(G,1,2)
+    assert_equal(list(list(p) for p in paths),[[1,2],[1,2]])
+
+def test_bidirectional_all_simple_paths_multigraph_with_cutoff():
+    G = nx.MultiGraph([(1,2),(1,2),(1,10),(10,2)])
+    paths = nx.bidirectional.all_simple_paths(G,1,2, cutoff=1)
+    assert_equal(list(list(p) for p in paths),[[1,2],[1,2]])
+
+def test_bidirectional_all_simple_paths_directed():
+    G = nx.DiGraph()
+    G.add_path([1,2,3])
+    G.add_path([3,2,1])
+    paths = nx.bidirectional.all_simple_paths(G,1,3)
+    assert_equal(list(list(p) for p in paths),[[1,2,3]])
+
+def test_bidirectional_all_simple_paths_empty():
+    G = nx.path_graph(4)
+    paths = nx.bidirectional.all_simple_paths(G,0,3,cutoff=2)
+    assert_equal(list(list(p) for p in paths),[])
+
+def hamiltonian_path(G,source):
+    source = next(G.nodes_iter())
+    neighbors = set(G[source])-set([source])
+    n = len(G)
+    for target in neighbors:
+        for path in nx.bidirectional.all_simple_paths(G,source,target):
+            if len(path) == n:
+                yield path
+
+def test_bidirectional_hamiltonian_path():
+    from itertools import permutations
+    G=nx.complete_graph(4)
+    paths = [list(p) for p in hamiltonian_path(G,0)]
+    exact = [[0]+list(p) for p in permutations([1,2,3],3) ]
+    assert_equal(sorted(paths),sorted(exact))
+
+def test_bidirectional_cutoff_zero():
+    G = nx.complete_graph(4)
+    paths = nx.bidirectional.all_simple_paths(G,0,3,cutoff=0)
+    assert_equal(list(list(p) for p in paths),[])
+    paths = nx.bidirectional.all_simple_paths(nx.MultiGraph(G),0,3,cutoff=0)
+    assert_equal(list(list(p) for p in paths),[])
+
+@raises(nx.NetworkXError)
+def test_bidirectional_source_missing():
+    G = nx.Graph()
+    G.add_path([1,2,3])
+    paths = list(nx.bidirectional.all_simple_paths(nx.MultiGraph(G),0,3))
+
+@raises(nx.NetworkXError)
+def test_bidirectional_target_missing():
+    G = nx.Graph()
+    G.add_path([1,2,3])
+    paths = list(nx.bidirectional.all_simple_paths(nx.MultiGraph(G),1,4))
+
+def test_bidirectional_shortest_simple_paths():
+    G = cnlti(nx.grid_2d_graph(4, 4), first_label=1, ordering="sorted")
+    paths = nx.bidirectional.all_shortest_paths(G, 1, 12)
+    assert_equal(next(paths), [1, 2, 3, 4, 8, 12])
+    assert_equal(next(paths), [1, 5, 9, 10, 11, 12])
+    assert_equal([len(path) for path in nx.bidirectional.all_simple_paths(G, 1, 12)],
```

Comment:
in this case I would leave it as it is, I do not think that readability is increased by splitting the code, since it is a copy the original simple and shortest path tests... see also

215 def test_bidirectional_dijksta_restricted():

which does not meet pep8 too, by the way, there is an (r) missing ;)